applications/tari_console_wallet/src/main.rs

```diff
@@ -111,30 +124,188 @@ fn main_inner() -> Result<(), ExitCodes> {
     let wallet = Arc::new(RwLock::new(wallet));
     let grpc = crate::grpc::WalletGrpcServer::new(wallet.clone());
 
-    if !bootstrap.daemon_mode {
-        runtime.spawn(run_grpc(grpc, node_config.grpc_wallet_address));
-        let app = App::<CrosstermBackend<Stdout>>::new(
-            "Tari Console Wallet".into(),
-            &node_identity,
-            wallet,
-            node_config.network,
-        );
-        runtime.handle().enter(|| ui::run(app))?;
-        println!("The wallet is shutting down.");
-        info!(
-            target: LOG_TARGET,
-            "Termination signal received from user. Shutting wallet down."
-        );
-        // TODO: Shutdown wallet
-        Ok(())
-    } else {
-        println!("Starting grpc server");
-        runtime
-            .block_on(run_grpc(grpc, node_config.grpc_wallet_address))
-            .map_err(ExitCodes::GrpcError)?;
-        println!("Shutting down");
-        Ok(())
+    let wallet_mode = match (bootstrap.daemon_mode, bootstrap.input_file) {
+        (false, None) => WalletMode::Tui,
+        (true, None) => WalletMode::Grpc,
+        (false, Some(path)) => WalletMode::Script(path),
+        (true, Some(_)) => WalletMode::Invalid,
+    };
+
+    match wallet_mode {
+        WalletMode::Tui => tui_mode(runtime, grpc, node_config, node_identity, wallet),
+        WalletMode::Grpc => grpc_mode(runtime, grpc, node_config),
+        WalletMode::Script(path) => script_mode(runtime, path, wallet),
+        WalletMode::Invalid => Err(ExitCodes::InputError(
+            "Can't run GRPC and file input mode simultaneously!".to_string(),
+        )),
+    }
+}
+
+fn script_mode(mut runtime: Runtime, path: PathBuf, wallet: WalletThreadSafe) -> Result<(), ExitCodes> {
```

Comment:
The script might also be a once off string

Reply:
sounds good, instead of an `input_file` give it an `input_command`